arch/x86/core/intel64/locore.S

```diff
@@ -59,7 +59,7 @@ __start:
 	movw %ax, %fs
 	movw %ax, %gs
 
-	movl $(exception_stack + CONFIG_EXCEPTION_STACK_SIZE), %esp
+	movl $(_exception_stack + CONFIG_EXCEPTION_STACK_SIZE), %esp
```

Comment:
we are moving away from declaring private globals with the prefix "_", it violates MISRA-C at least as we currently interpret it.

Suggest calling this z_x86_exception_stack since it seems to be only referenced within arch/x86.
Yes, there's *tons* of housecleaning that needs to be done already in arch/x86/ related to this, because of that, not gating for this PR.

Reply:
Yeah, I did that for parity with `_interrupt_stack`.  The x86 stuff really does need a naming workover (if only for consistency's sake) -- I'm as guilty as the rest of contributing to the problem, no doubt about it ..